pkg/logs/internal/launchers/file/launcher.go

```diff
@@ -427,3 +436,13 @@ func (s *Launcher) createTailer(file *tailer.File, outputChan chan *message.Mess
 func (s *Launcher) createRotatedTailer(t *tailer.Tailer, file *tailer.File, pattern *regexp.Regexp) *tailer.Tailer {
 	return t.NewRotatedTailer(file, decoder.NewDecoderFromSourceWithPattern(file.Source, pattern))
 }
+
+func CheckProcessTelemetry(stats *util.ProcessFileStats) {
+	// Log a warning if the ratio between the Agent's open files to the OS file limit is > 0.9, log an error if OS file limit is reached
+	if stats.AgentOpenFiles/stats.OsFileLimit > 0.9 {
+		log.Warnf("Agent process is close to OS file limit of %v. Agent process currently has %v files open.", stats.OsFileLimit, stats.AgentOpenFiles)
+	} else if stats.AgentOpenFiles/stats.OsFileLimit >= 1 {
+		log.Errorf("Agent process is reaching OS open file limit: %v. This may be preventing log files from being tailed by the Agent. Consider increasing OS file limit.", stats.OsFileLimit)
```

Comment:
```suggestion
		log.Errorf("Agent process has reached the OS open file limit: %v. This may be preventing log files from being tailed by the Agent and could interfere with the basic functionality of the Agent. OS file limit must be increased.", stats.OsFileLimit)
```

Added a bit stronger wording here, this is a nearly-fatal condition for the agent to be in, it may not be able to open new connections to the DD intake for example.